Src/Newtonsoft.Json/JsonReader.cs

```diff
@@ -1109,5 +987,50 @@ public virtual void Close()
             _tokenType = JsonToken.None;
             _value = null;
         }
+
+        internal void ReadAndAssert()
+        {
+            if (!Read())
+            {
+                throw JsonSerializationException.Create(this, "Unexpected end.");
```

Comment:
Change the error message to "Unexpected end when reading JSON". I only had just "unexpected end" because I was lazy writing messages.